tools/DevDiagnostics/DevHome.DevDiagnostics/Controls/ThemeAwareWindow.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.UI;
+using Microsoft.UI.Dispatching;
+using Microsoft.UI.Windowing;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Media;
+using Windows.UI;
+using Windows.UI.ViewManagement;
+using WinUIEx;
+
+namespace DevHome.DevDiagnostics.Controls;
+
+/* This class encapsulates the theme-awareness logic, including support for custom chrome buttons.
+ * The main BarWindow derives from ThemeAwareWindow. All internal tool windows and any other owned windows
+ * must follow the theme of the main BarWindow. To achieve this:
+ * 1. Derive the tool window from ThemeAwareWindow.
+ * 2. Define a grid or panel for the icon and title (because this window sets ExtendsContentIntoTitleBar
+ *    and HideIconAndSystemMenu).
+ * 3. Add the tool window to the list of related windows for the BarWindow by calling AddRelatedWindow.
+ * 4. Remove the tool window from the list of related windows for the BarWindow when the tool window is closed.
```

Comment:
Both of these seem like candidates for code reusability.  Maybe the base class should do this in some manner.

Reply:
The issue here is that both the main BarWindow and the tool windows derive from the same base class. You could argue that I should separate the common theme behavior from the related-window behavior - and I wouldn't disagree, but we can't derive a window class from multiple classes.

Reply:
You could hardcode the base class to recognize BarWindow, since it is only "one" whereas the others are dynamically "many".  Also, you could look into Generics and include the BarWindow class as a type argument to do something similar but less hardcoded.

Reply:
Yes, I see this PR as temporary. It fixes the issues, but I will update it, incorporating this feedback, once we move to WAS 1.6, per new issue: #3803